cdc/sink/mysql.go

```diff
@@ -390,15 +391,22 @@ func querySQLMode(ctx context.Context, db *sql.DB) (sqlMode string, err error) {
 }
 
 // check whether the target charset is supported
-func checkCharsetSupport(ctx context.Context, db *sql.DB, _ string) (bool, error) {
+func checkCharsetSupport(ctx context.Context, db *sql.DB, charsetName string) (bool, error) {
 	var (
-		charset          string
+		charSet          string
 		description      string
 		defaultCollation string
 		maxLen           int
 	)
-	err := db.QueryRowContext(ctx, "show character set where charset = 'gbk';").
-		Scan(&charset, &description, &defaultCollation, &maxLen)
+
+	// validate charsetName
+	_, err := charset.GetCharsetInfo(charsetName)
+	if err != nil {
+		return false, errors.Trace(err)
+	}
+
+	err = db.QueryRowContext(ctx, "show character set where charset = '"+charsetName+"';").
```

Comment:
Prefer `SELECT [only columns we need] FROM INFORMATION_SCHEMA.CHARACTER_SETS WHERE xxx` to prevent possible incompatible changes in downstream databases, e.g., `show character set` returns more than 4 columns.